src/foam/nanos/pm/PM.js

```diff
@@ -23,6 +23,15 @@ foam.CLASS({
   ids: [ 'classType', 'name', 'startTime' ],
 
   properties: [
+    {
+      class: 'String',
+      name: 'id',
+      javaFactory: `
+        if ( getClassType() != null )
+          return getClassType().getId();
+        return null;
+      `
+    },
     {
       class: 'Class',
```

Comment:
remove this please, but keep the constructors for backward compatibility, and set the Id to this property as it is now.